OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -128,4 +128,17 @@ void Frame::extendRealizeTopology(SimTK::State& s) const
 {
     Super::extendRealizeTopology(s);
     groundTransformIndex = getCacheVariableIndex("ground_transform");
+}
+void Frame::extendRealizeVelocity(const SimTK::State& s) const
+{   
+    //Evaluate the lazy cache for the groundTransform so it doesn't get
+    //needlesly calculated by each one of the threads while realizing
+    //dynamics. (Also for thread-safety reasons)
+    Super::extendRealizeVelocity(s);
+
+    SimTK::Value<SimTK::Transform>::downcast(getSystem().getDefaultSubsystem().
+        updCacheEntry(s, groundTransformIndex)).upd() = calcGroundTransform(s);
+    
+    getSystem().getDefaultSubsystem().
+        markCacheValueRealized(s, groundTransformIndex);
```

Comment:
@sherm1 mentioned in an outdated diff:

> This should not be done in realizeVelocity(). It should be done in realizePosition() because it only involves body positions. It will be repeatedly recalculated unnecessarily if velocities get updated while positions remain fixed (and that does happen).